apps/glusterfs/volume_entry.go

```diff
@@ -304,6 +304,11 @@ func (v *VolumeEntry) Create(db *bolt.DB,
 			break
 		}
 	}
+
+	if err != nil {
+		return err
+	}
+
```

Comment:
I think the correct fix will probably be to add what you have, but then move the logic of _ErrNoSpace_ to v.allocBricksInCluster(), then change line 300 to say something like: if err == ErrNoSpace, then continue, else get out!